infra/azure/__main__.py

```diff
@@ -394,3 +395,187 @@ def get_kubeconfig(
+    version="0.45.12",
+    repository_opts=RepositoryOptsArgs(
+        repo="https://argoproj.github.io/argo-helm",
+    ),
+    value_yaml_files=[
+        FileAsset("./k8s/argo_workflows/values.yaml"),
+    ],
+    values={
+        "controller": {
+            "workflowNamespaces": [argo_workflows_ns.metadata.name]
+        },
+        "server": {
+            "ingress": {
+                "hosts": [argo_url],
+                "tls":[
+                    {
+                        "secretName": "argo-ingress-tls-letsencrypt",
+                        "hosts": [argo_url],
+                    }
+                ],
+            },
+            "sso": {
+                "enabled": True,
+                "issuer": Output.concat("https://login.microsoftonline.com/", config.require_secret("entra_tenant_id"), "/v2.0"),
+                "redirectUri": Output.concat(
+                    "https://", argo_url, "/oauth2/callback"
+                ),
+            },
+        },
+    },
+    opts=ResourceOptions(
+        provider=k8s_provider,
+        depends_on=[argo_minio_secret, argo_sso_secret, argo_server_ns, argo_workflows_ns, managed_cluster],
+    ),
+)
+
+argo_workflows_admin_sa = ServiceAccount(
+    "argo-workflows-admin-sa",
+    metadata=ObjectMetaArgs(
+        name="argo-workflows-admin-sa",
+        namespace=argo_server_ns.metadata.name,
+        annotations={
+            "workflows.argoproj.io/rbac-rule": Output.concat("'", config.require_secret("admin_entra_group_id"),"'", " in groups"),
+            "workflows.argoproj.io/rbac-rule-precedence": "2",
+        }
+    ),
+    opts=ResourceOptions(
+        provider=k8s_provider,
+        depends_on=[argo_workflows],
+    ),
+)
+
+argo_workflows_admin_sa_token = Secret(
+    "argo-workflows-admin-sa-token",
+    metadata=ObjectMetaArgs(
+        name="argo-workflows-admin-sa.service-account-token",
+        namespace=argo_server_ns.metadata.name,
+        annotations={
+            "kubernetes.io/service-account.name": argo_workflows_admin_sa.metadata.name,
+        }
+    ),
+    type="kubernetes.io/service-account-token",
+    opts=ResourceOptions(
+        provider=k8s_provider,
+        depends_on=[argo_workflows_admin_sa],
+    ),
+)
+
+argo_workflows_admin_role = Role(
+    "argo-workflows-admin-role",
+    metadata=ObjectMetaArgs(
+        name="argo-workflows-admin-role",
+        namespace=argo_server_ns.metadata.name,
+    ),
+    rules=[
+        PolicyRuleArgs(
+            api_groups=[""],
+            resources=["events", "pods", "pods/log"],
+            verbs=["get", "list", "watch"],
+        ),
+        PolicyRuleArgs(
+            api_groups=["argoproj.io"],
+            resources=["cronworkflows", "eventsources", "workflows", "workflows/finalizers", "workflowtaskresults", "workflowtemplates", "clusterworkflowtemplates"],
+            verbs=["create", "delete", "deletecollection", "get", "list", "patch", "watch", "update"],
+        ),
+    ],
```

Comment:
It would be good to have a comment here with a reference to the docs. Good to know why we have chosen these rules.